### PR TITLE
Implement support for set_turn_to_ice attribute class

### PR DESCRIPTION
**Pull request by M60TM**
It exists in [Hidden dev attribute](https://github.com/rafradek/TF2-Hidden-Attributes). But `set_turn_to_ice` is staging-only attribute. So I decided to make implement support for that.

### CODE REVIEW

File: gamedata/tf2.attribute_support.txt [hunk 1]
```diff
@@ -270,6 +334,11 @@
 				"windows"	"481"
 				"linux"		"488"
 			}
+			"CBasePlayer::Event_Killed()"
```
Comment (nosoop) [inline gamedata/tf2.attribute_support.txt]: I'd want to use a detour on `CTFPlayer::Event_Killed` instead; virtual hooks tend to be problematic at times.

File: scripting/tf2attribute_support.sp [hunk 2]
```diff
@@ -1,6 +1,6 @@
 /**
  * TF2 Attribute Extended Support plugin
- * 
+ *
```
Comment (nosoop) [inline scripting/tf2attribute_support.sp]: Whitespace changes make the diff noisy; I'd like them reverted.

File: scripting/tf2attribute_support.sp [hunk 3]
```diff
@@ -33,11 +33,11 @@ public Plugin myinfo = {
 
 Handle g_DHookBaseEntityGetDamage;
 Handle g_DHookWeaponSendAnim;
-Handle g_DHookGrenadeInit;
```
Comment (nosoop) [inline scripting/tf2attribute_support.sp]: Unrelated change; while it isn't used anymore, I'd want this committed separately.

File: scripting/tf2attribute_support.sp [hunk 4]
```diff
@@ -1040,6 +1056,36 @@ MeterType GetItemMeterType(int item) {
 	return Meter_Invalid;
 }
 
+MRESReturn DHookEvent_KilledPre(int client, DHookParam hParams)
```
Comment (nosoop) [inline scripting/tf2attribute_support.sp]: With a detour change, I think `g_bShouldTurnToIce[client]` should always be set to false at the start of `DHookEvent_KilledPre`.  The subsequent code could be simplified to:

```sourcepawn
g_bShouldTurnToIce[client] = IsValidEntity(weapon) && TF2Attrib_HookValueInt(0, "set_turn_to_ice", weapon) != 0;
```

**Review by nosoop — CHANGES_REQUESTED** [review]
Looks like `set_turn_to_ice` is actually used, but only for ragdolls spawned by the Dead Ringer; the normal live ragdoll code only checks for `freeze_backstab_victim` specifically on a backstab.

Thank you for the contribution &mdash; I'd like some adjustments made.  I'll try to get to it in a few hours unless you'd like to make them yourself.

**Comment by M60TM** [comment]
Thank you for review. But I don't have enough time to change them myself. I'd really appreciate it if you could do it for me.
